app/js/lib/foxbox-qr.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+ /* global cordova, cordovaHTTP */
+
+// Expecting URLs of the form:
+// https://192-168-0-16.0123456789abcdef0123456789abcdef.plex.direct:12346/
+// where the first part is a local IPv4 address and the second part is a 32-char
+// alphanumeric string, representing the first 128 bits of the sha256 hash of
+// the root certificate with which the local server's TLS certificate is signed,
+// in hexadecimal representation.
+const FINGERPRINT_PART = 1;
+
+export default class Qr {
+  constructor() {
+    if (cordova && cordova.plugins &&
+        cordova.plugins.barcodeScanner &&
+        cordovaHTTP && cordovaHTTP.acceptHss) {
+      this.supported = true;
+    } else {
+      this.supported = true;
+      console.warn('QR code based discovery disabled (Cordova plugins not available)');
```

Comment:
I'm not fan of a warning that'll show all the time in the console even if running from a browser.
Can you return earlier from the constructor if we're not in a Cordova app and then show the warning only if in a Cordova app and something went wrong?